test/manipulation.test.js

```diff
@@ -1686,7 +1700,73 @@ describe('manipulation', function() {
     });
 
     // TODO: implement destroy with filtered set
-    it('should destroy filtered set of records');
+    it.skip('should destroy filtered set of records');
+
+    it('should not delete using condition from filter', function(done) {
+      Person.destroyAll({where: {name: 'John'}}, function(err, info) {
+        should.exist(err);
+        err.should.match(/Filter object detected/);
+        Person.find({fields: 'name'}, function(err, users) {
+          should.not.exist(err);
```

Comment:
This is an anti-pattern I have been fighting for years.

When `Person.find` fails, this assertions says something along the lines "Expected {Error} to not exist" and shows a stack trace pointing to this very line, throwing the original stack trace away.

A better solution is to pass `err` to test callback. That way the test fails with the original error triggered by `Person.find` and provides all information from the original error (message, stack trace, error code, etc.)

```js
if (err) return done(err);
```